### PR TITLE
Modified storage related functions to use new structs

### LINKED ISSUE #1289: Modify Storage to use new structs
- [ ] Add missing fields
- [ ] Update all functions that refer to Storage struct to new structs (machine readable)

### PR DESCRIPTION
**Pull request by mik-dass**
getMachineReadableFormat() and getMachineReadableFormatForList() were introduced for helping in making machine readable output for storage.

fixes #1289 

How to test:

Test all storage related commands

### CODE REVIEW

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -222,10 +212,14 @@ func ListUnmounted(client *occlient.Client, applicationName string) ([]StorageIn
 			}
 			storageName := getStorageFromPVC(&pvc)
 			storageSize := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
-			storageList = append(storageList, StorageInfo{
-				Name: storageName,
-				Size: storageSize.String(),
-				Path: "",
+			storageList = append(storageList, Storage{
```
Comment (girishramnani) [inline pkg/storage/storage.go]: can we have a `NewStorage` or `NewMachineReadableStructForStorage` to hide the meta and spec here?

File: pkg/component/component.go [hunk 2]
```diff
@@ -35,14 +35,14 @@ const componentNameMaxLen = -1
...
-	URLs               []urlpkg.URL          `json:"url,omitempty"`
-	Env                []corev1.EnvVar       `json:"environment,omitempty"`
-	Storage            []storage.StorageInfo `json:"storage,omitempty"`
-	LinkedComponents   map[string][]string   `json:"linkedComponents,omitempty"`
-	LinkedServices     []string              `json:"linkedServices,omitempty"`
+	ComponentName      string              `json:"componentName,omitempty"`
+	ComponentImageType string              `json:"type,omitempty"`
+	Path               string              `json:"source,omitempty"`
+	URLs               []urlpkg.URL        `json:"url,omitempty"`
+	Env                []corev1.EnvVar     `json:"environment,omitempty"`
```
Comment (girishramnani) [inline pkg/component/component.go]: do we need these json tags @mik-dass ?

**Comment by mik-dass** [comment]
@girishramnani Fixed :)

**Comment by girishramnani** [comment]
Also if the codeclimate issues are reasonable then can we solve them

**Comment by mik-dass** [comment]
@girishramnani I guess we already have e2e tests for the json machine readable part and my changes are internal changes :)

**Comment by girishramnani** [comment]
pinging @amitkrout for test related stuff

**Comment by mik-dass** [comment]
> @mik-dass Please address the missing UTs

I have added the UTs for those functions, please have a look again :)

**Pull request merged**